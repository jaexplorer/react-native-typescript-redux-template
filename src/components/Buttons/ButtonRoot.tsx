import React, { FC } from 'react';
import { TouchableOpacity } from 'react-native';
import { Style } from '../../lib/Types';
import { useTheme } from '../../lib/Theme';
import stylesCreator from './ButtonStyles';

const ButtonRoot: FC<ButtonRootProps> = ({
  children,
  onPress = () => {},
  disabled = false,
  style = {},
}) => {
  const [styles] = useTheme(stylesCreator, { disabled });
  return (
    <TouchableOpacity
      style={[styles.buttonRootContainer, style]}
      activeOpacity={disabled ? 0.4 : 0.2}
      onPress={disabled ? () => {} : onPress}
    >
      {children}
    </TouchableOpacity>
  );
};

export interface ButtonRootProps {
  onPress?: () => void;
  disabled?: boolean;
  rnTouchable?: boolean;
  label?: string;
  bold?: boolean;
  style?: Style;
}

export default ButtonRoot;
