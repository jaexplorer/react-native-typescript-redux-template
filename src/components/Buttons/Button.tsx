import React, { FC } from 'react';
import Text from '../Text';
import { Style } from '../../lib/Types';
import ButtonRoot, { ButtonRootProps } from './ButtonRoot';
import { useTheme } from '../../lib/Theme';
import stylesCreator from './ButtonStyles';

const Button: FC<ButtonProps> = ({
  onPress = () => {},
  disabled = false,
  primary = true,
  children,
  style = {},
  textStyle = {},
}) => {
  const [styles] = useTheme(stylesCreator, { primary });

  return (
    <ButtonRoot style={[styles.buttonContainer, style]} onPress={onPress} disabled={disabled}>
      <Text style={[styles.buttonText, textStyle]}>{children}</Text>
    </ButtonRoot>
  );
};

export interface ButtonProps extends ButtonRootProps {
  textStyle?: Style;
  children: string;
  primary?: boolean;
}

export default Button;
