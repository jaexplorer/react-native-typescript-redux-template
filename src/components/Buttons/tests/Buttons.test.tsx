import React from 'react';
import { View } from 'react-native';
import { render } from '@testing-library/react-native';
import Button from '../Button';
import ButtonRoot from '../ButtonRoot';

test('Does Button render test', () => {
  const { baseElement } = render(<Button>Test</Button>);
  expect(baseElement).toMatchSnapshot();
});

test('Does ButtonRoot render test', () => {
  const { baseElement } = render(
    <ButtonRoot>
      <View />
    </ButtonRoot>,
  );
  expect(baseElement).toMatchSnapshot();
});
