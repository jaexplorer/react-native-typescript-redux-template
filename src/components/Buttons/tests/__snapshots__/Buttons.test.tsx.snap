// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Does Button render test 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <
      activeOpacity={0.2}
      style={
        Array [
          Object {
            "opacity": 1,
          },
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#4D4D4D",
              "borderColor": "#999999",
              "borderRadius": 100,
              "borderWidth": 0,
              "justifyContent": "center",
              "minWidth": 128.57142857142856,
              "padding": 10.714285714285714,
            },
            Object {},
          ],
        ]
      }
    >
      <Text
        maxFontSizeMultiplier={1}
        style={
          Array [
            Object {
              "bottom": -5,
              "color": "#333333",
              "fontSize": 15.714285714285714,
              "marginBottom": 0,
              "textAlign": "center",
              "textAlignVertical": "center",
            },
            Array [
              Object {
                "color": "#333333",
                "fontFamily": "",
                "fontSize": 15.714285714285714,
                "paddingLeft": 21.428571428571427,
                "paddingRight": 21.428571428571427,
              },
              Object {},
            ],
          ]
        }
      >
        Test
      </Text>
    </>
  </View>
</View>
`;

exports[`Does ButtonRoot render test 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <
      activeOpacity={0.2}
      style={
        Array [
          Object {
            "opacity": 1,
          },
          Object {},
        ]
      }
    >
      <View />
    </>
  </View>
</View>
`;
